src/libcore/char/convert.rs

```diff
@@ -99,7 +99,7 @@ pub fn from_u32(i: u32) -> Option<char> {
 #[inline]
 #[stable(feature = "char_from_unchecked", since = "1.5.0")]
 pub unsafe fn from_u32_unchecked(i: u32) -> char {
-    transmute(i)
+    if cfg!(debug_assertions) { char::from_u32(i).unwrap() } else { transmute(i) }
```

Comment:
`from_u32` ends up calling `from_u32_unchecked` so won't this result in an infinite recursion?

Reply:
*oops* looks like I did not dig deep enough when I looked at the call graph here.